lib/operations/estimated_document_count.js

```diff
@@ -2,32 +2,57 @@
 
 const Aspect = require('./operation').Aspect;
 const defineAspects = require('./operation').defineAspects;
-const CommandOperation = require('./command');
-const buildCountCommand = require('./common_functions').buildCountCommand;
-const handleCallback = require('../utils').handleCallback;
-
-class EstimatedDocumentCountOperation extends CommandOperation {
-  constructor(collection, options) {
-    super(collection.s.db, options, collection);
+const CommandOperationV2 = require('./command_v2');
+
+class EstimatedDocumentCountOperation extends CommandOperationV2 {
+  constructor(collection, query, options) {
+    if (typeof options === 'undefined') {
+      options = query;
+      query = undefined;
+    }
+
+    super(collection, options);
+    this.collectionName = collection.s.namespace.collection;
```

Comment:
for future thought: maybe we should capture the namespace within the `CommandOperation`?

Reply:
Seems reasonable, I might do it with the upcoming `FindOperation` if that' okay with you?